contrib/unbound-cert-setup.sh

```diff
@@ -0,0 +1,141 @@
+#!/bin/bash
+
+# Generally based on /usr/sbin/unbound-control-setup but adapted to catch
+# up to ~2010. You know, x509v3, secp384r1, AKIs, stuff like that.
+
+# directory for files
+DESTDIR=/etc/unbound
+
+# validity period for certificates
+DAYS=7200
+
+# hash algorithm
+HASH=sha256
+
+# base name for unbound CA keys
+CA_BASE=unbound_ca_ec
+
+# base name for unbound server keys
+SVR_BASE=unbound_server_ec
+
+# base name for unbound-control keys
+CTL_BASE=unbound_control_ec
+
+# we want -rw-r----- access (say you run this as root: grp=yes (server), all=no).
+umask 0027
+
+# end of options
+
+# functions:
+error ( ) {
+        echo "$0 fatal error: ${1}"
+        exit 1
+}
+
+# go!:
+echo "setup in directory ${DESTDIR}"
+cd "${DESTDIR}" || error "could not cd to ${DESTDIR}"
+
+# create certificate keys; do not recreate if they already exist.
+if test -f "${CA_BASE}.key"; then
+        echo "${CA_BASE}.key exists"
+else
+        echo "generating ${CA_BASE}.key"
+        openssl ecparam -genkey -name secp384r1 > ${CA_BASE}.key || error "could not gen ecdsa"
+fi
+if test -f "${SVR_BASE}.key"; then
+        echo "${SVR_BASE}.key exists"
+else
+        echo "generating ${SVR_BASE}.key"
+        openssl ecparam -genkey -name secp384r1 > ${SVR_BASE}.key || error "could not gen ecdsa"
+fi
+if test -f "${CTL_BASE}.key"; then
+        echo "${CTL_BASE}.key exists"
+else
+        echo "generating ${CTL_BASE}.key"
+        openssl ecparam -genkey -name secp384r1 > ${CTL_BASE}.key || error "could not gen ecdsa"
+fi
+
+# create self-signed cert CSR for server
+cat > ca_request.cfg <<EOCAConfig
+[req]
+prompt                 = no
+distinguished_name     = req_distinguished_name
+x509_extensions        = req_v3_extensions
+
+[req_distinguished_name]
+commonName             = unbound-ca
+
+[req_v3_extensions]
+subjectKeyIdentifier   = hash
+authorityKeyIdentifier = keyid:always,issuer:always
+basicConstraints       = CA:true
+EOCAConfig
+
+test -f ca_request.cfg || error "could not create ca_request.cfg"
+
+echo "create ${CA_BASE}.pem (self signed certificate)"
+openssl req -key "${CA_BASE}.key" -config ca_request.cfg  -new -x509 -days "${DAYS}" -out "${CA_BASE}.pem" || error "could not create ${CA_BASE}.pem"
+# create trusted usage pem
+openssl x509 -in "${CA_BASE}.pem" -addtrust serverAuth -out "${CA_BASE}_trust.pem"
```

Comment:
Why can't the serverAuth KU be specified in the ca_request.cfg? Seems like this two-step "make a CSR, sign the cert" could be a piped single-step like the two signings below. Why can't it?